src/Camille.RiotGames/src/Util/RegionalRequester.cs

```diff
@@ -37,13 +36,44 @@ public class RegionalRequester
         /// </summary>
         private readonly HttpClient _client = new HttpClient();
 
+        /// <summary>
+        /// The region currently being used.
+        /// </summary>
+        private string _region;
+
         public RegionalRequester(IRiotGamesApiConfig config, string route)
         {
             _config = config;
             _appRateLimit = new RateLimit(RateLimitType.Application, config);
+            _region = route;
+
+            // If the region is to be used in the url as a subdomain
+            if (_config.ApiCallRegionConfig == RegionConfig.InUrlAsSubdomain)
+            {
+                _client.BaseAddress = new Uri(string.Format($"https://{config.ApiUrl}", route));
+            }
+            // If the region is to be used in the url as a query parameter or header
+            else
+            {
+                var uri = $"https://{config.ApiUrl}";
+                if (uri.Contains("{0}"))
+                {
+                    string.Format(uri, "");
+                }
+                _client.BaseAddress = new Uri(uri);
+            }
 
-            _client.BaseAddress = new Uri($"https://{route}{RiotRootUrl}");
-            _client.DefaultRequestHeaders.Add(RiotKeyHeader, config.ApiKey);
+            // The header for the region, if requested
+            if (_config.ApiCallRegionConfig == RegionConfig.InHeader)
+            {
+                _client.DefaultRequestHeaders.Add(config.RegionHeaderKey, route);
+            }
+
+            // The API key is only needed for riot's API, otherwise it is assumed to be a keyed proxy
+            if (_config.ApiUrl.Contains("riotgames.com"))
+            {
+                _client.DefaultRequestHeaders.Add(RiotKeyHeader, config.ApiKey);
+            }
```

Comment:
Would prefer to not do any branching based on searching the ApiUrl - could either make this another config field like `includeRiotApiHeader` --

Or if the user is using a proxy they could just set the config.ApiKey to a meaningless value and the proxy can ignore it (maybe a bit sloppy but should work fine)